### PR TITLE
crdt: remoteDelete return -1 on no deletion

### LINKED ISSUE #7: Handle returning line number on missing deletion character
We may return a larger line number which may be out of range in CodeMirror. Handle by returning -1?

### PR DESCRIPTION
**Pull request by shanthanu9**
Fix #7

### CODE REVIEW

File: assets/js/crdt.js [hunk 1]
```diff
@@ -365,13 +369,16 @@ class CRDT {
     }
 
     /**
-     * Delete the `character` received from the CRDT structure if present
+     * Delete the `character` received from the CRDT structure if present.
+     * Returns -1 if character is not deleted (i.e. when already locally deleted)
```
Comment (rohit-mp) [inline assets/js/crdt.js]: when locally deleted or already deleted by someone else before whose operation reached first.
I guess the explanation is not needed.

**Comment by rohit-mp** [comment]
https://github.com/rohit-mp/codeshare/blob/0934bc22e52a0e42872b1debce2ee51e0d51c1d9/assets/js/crdt.js#L380-L389
Just adding an else statement to the outer `if` to return -1 would be sufficient I guess?
Similar for `remoteDeleteNewline` too

File: assets/js/crdt.js (hunk 1 above)
Comment (shanthanu9) [reply assets/js/crdt.js]: Right.

**Comment by shanthanu9** [comment]
But that won't work no? If the character is at line 3, the outer if will return false for line 1 and 2. I guess you mean replace break on line 388 with `return -1`?
It's fine I guess. Having an extra variable makes it more explicit anyways

**Pull request merged**